kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/IcebergWriterCompatV1Suite.scala

```diff
@@ -128,6 +130,108 @@ class IcebergWriterCompatV1Suite extends DeltaTableWriteSuiteBase with ColumnMap
+          .containsEntry(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY, "col-1")
+
+        // Add a new column and verify column mapping info set correctly
+        val updatedSchema = if (cmInfoPopulated) {
+          initialMetadata.getSchema
+            .add(
+              "c2",
+              IntegerType.INTEGER,
+              FieldMetadata.builder()
+                .putLong(ColumnMapping.COLUMN_MAPPING_ID_KEY, 2)
+                .putString(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY, "col-2")
+                .build())
+        } else {
+          initialMetadata.getSchema
+            .add("c2", IntegerType.INTEGER)
+        }
+        createWriteTxnBuilder(Table.forPath(engine, tablePath))
+          .withSchema(engine, updatedSchema)
+          .build(engine)
+          .commit(engine, emptyIterable())
+        val updatedMetadata = getMetadata(engine, tablePath)
+        assertThat(updatedMetadata.getConfiguration)
+          .containsEntry(ColumnMapping.COLUMN_MAPPING_MAX_COLUMN_ID_KEY, "2")
+        assertThat(updatedMetadata.getSchema.get("c2").getMetadata.getEntries)
+          .containsEntry(ColumnMapping.COLUMN_MAPPING_ID_KEY, 2L.asInstanceOf[AnyRef])
+          .containsEntry(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY, "col-2")
+      }
+    }
+  }
+
+  Seq(true, false).foreach { isNewTable =>
+    test(s"Cannot set physicalName to something other than col-{fieldId}, isNewTable=$isNewTable") {
+      withTempDirAndEngine { (tablePath, engine) =>
+        if (!isNewTable) {
+          createEmptyTable(
+            engine,
+            tablePath,
+            new StructType().add("c1", IntegerType.INTEGER),
+            tableProperties = tblPropertiesIcebergWriterCompatV1Enabled)
+        }
+        val schemaToCommit = if (isNewTable) {
+          new StructType()
+            .add(
+              "c2",
+              IntegerType.INTEGER,
+              FieldMetadata.builder()
+                .putLong(ColumnMapping.COLUMN_MAPPING_ID_KEY, 1)
+                .putString(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY, "c2")
+                .build())
+        } else {
+          getMetadata(engine, tablePath)
+            .getSchema
+            .add(
+              "c2",
+              IntegerType.INTEGER,
+              FieldMetadata.builder()
+                .putLong(ColumnMapping.COLUMN_MAPPING_ID_KEY, 2)
+                .putString(ColumnMapping.COLUMN_MAPPING_PHYSICAL_NAME_KEY, "c2")
+                .build())
+        }
+        val e = intercept[KernelException] {
+          createWriteTxnBuilder(Table.forPath(engine, tablePath))
+            .withTableProperties(engine, tblPropertiesIcebergWriterCompatV1Enabled.asJava)
+            .withSchema(engine, schemaToCommit)
+            .build(engine)
+            .commit(engine, emptyIterable())
+        }
+        assert(e.getMessage.contains(
+          "IcebergWriterCompatV1 requires column mapping field physical names be equal to "
```

Comment:
We know what the full error will be right? and the computation of the extra (and nice) info is somewhat complex, so maybe check the full correct message is there.

Reply:
Yeah let me update it; I think I was thinking it would be kinda complex since we don't just format the column name, but other info as well -- but I think it shouldn't be too much to do

Reply:
Updated this test + the one in the unit test suite